pkg/graveler/ref/repository_iterator.go

```diff
@@ -4,99 +4,87 @@ import (
 	"context"
 	"errors"
 
-	"github.com/treeverse/lakefs/pkg/db"
 	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/kv"
 )
 
-type RepositoryIterator struct {
-	db        db.Database
-	ctx       context.Context
-	value     *graveler.RepositoryRecord
-	buf       []*graveler.RepositoryRecord
-	offset    string
-	fetchSize int
-	err       error
-	state     iteratorState
+type RepositoryIterator interface {
+	Next() bool
+	SeekGE(id graveler.RepositoryID)
+	Value() *graveler.RepositoryRecord
+	Err() error
+	Close()
 }
 
-func NewRepositoryIterator(ctx context.Context, db db.Database, fetchSize int) *RepositoryIterator {
-	return &RepositoryIterator{
-		db:        db,
-		ctx:       ctx,
-		fetchSize: fetchSize,
-		buf:       make([]*graveler.RepositoryRecord, 0, fetchSize),
-	}
+type KVRepositoryIterator struct {
+	ctx   context.Context
+	it    kv.MessageIterator
+	err   error
+	value *graveler.RepositoryRecord
+	store kv.Store
 }
 
-func (ri *RepositoryIterator) Next() bool {
-	if ri.err != nil {
-		return false
+func NewKVRepositoryIterator(ctx context.Context, store *kv.StoreMessage) (*KVRepositoryIterator, error) {
+	it, err := kv.NewPrimaryIterator(ctx, store.Store, (&graveler.RepositoryData{}).ProtoReflect().Type(), graveler.RepoPartition(), []byte(graveler.RepoPath("")), kv.IteratorOptionsAfter([]byte{}))
+	if err != nil {
+		return nil, err
 	}
+	return &KVRepositoryIterator{
+		it:    it,
+		store: store.Store,
+		ctx:   ctx,
+	}, nil
+}
 
-	ri.maybeFetch()
-
-	// stage a value and increment offset
-	if len(ri.buf) == 0 {
+func (ri *KVRepositoryIterator) Next() bool {
+	if ri.Err() != nil {
 		return false
 	}
-	ri.value = ri.buf[0]
-	ri.buf = ri.buf[1:]
-	ri.offset = string(ri.value.RepositoryID)
-	return true
-}
 
-func (ri *RepositoryIterator) maybeFetch() {
-	if ri.state == iteratorStateDone {
-		return
+	if !ri.it.Next() {
+		ri.value = nil
+		return false
 	}
-	if len(ri.buf) > 0 {
-		return
+	e := ri.it.Entry()
+	if e == nil {
+		ri.err = graveler.ErrNotFound
```

Comment:
I don't think `graveler.ErrNotFound` is appropriate here

Reply:
Done